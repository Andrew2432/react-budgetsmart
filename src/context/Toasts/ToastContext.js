import { createContext } from 'react';

// Create a context for toasts and export it
const ToastContext = createContext();

export default ToastContext;
