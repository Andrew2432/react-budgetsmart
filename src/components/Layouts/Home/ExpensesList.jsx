import React, { useContext, useEffect } from 'react';
import {
  CardPanel,
  Collection,
  CollectionItem,
  Row,
  Col,
  Button,
} from 'react-materialize';
import ExpenseItem from './ExpenseItem';
import ConfirmModal from '../Utils/ConfirmModal';
import ExpenseContext from '../../../context/Expense/expenseContext';
import ToastContext from '../../../context/Toasts/ToastContext';

const ExpensesList = () => {
  const expenseContext = useContext(ExpenseContext);
  const toastContext = useContext(ToastContext);

  const { setToast } = toastContext;
  const {
    expenses,
    clearAllExpense,
    removeToastMode,
    toastMode,
  } = expenseContext;

  useEffect(() => {
    switch (toastMode) {
      case 'edit':
        setToast({ type: 'success', text: 'Updated successfully' });
        break;
      case 'delete':
        setToast({ type: 'success', text: 'Deleted successfully' });
        break;
      case 'clear':
        setToast({ type: 'success', text: 'Cleared successfully' });
        break;
      default:
        break;
    }
    removeToastMode();
    // eslint-disable-next-line
  }, [toastMode]);

  const createExpenseItem = (expense) => (
    <ExpenseItem {...expense} key={expense.id} />
  );

  return (
    <CardPanel className="indigo accent-2">
      <h4>Your Expenses</h4>
      <Collection className="black-text">
        {expenses.length > 0 ? (
          expenses.map(createExpenseItem)
        ) : (
          <CollectionItem>No expenses added. Enter an expense</CollectionItem>
        )}
      </Collection>
      {expenses.length > 0 && (
        <Row>
          <Col>
            <ConfirmModal
              onClick={clearAllExpense}
              trigger={
                <Button className="purple darken-2">Clear Expenses</Button>
              }
              message="Are you sure you want to delete all expenses?"
            />
          </Col>
        </Row>
      )}
    </CardPanel>
  );
};

export default ExpensesList;
