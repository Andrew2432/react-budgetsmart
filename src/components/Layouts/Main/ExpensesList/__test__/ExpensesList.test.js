import TestRenderer from 'react-test-renderer';
import React from 'react';
import ExpensesList from '../ExpensesList';
import ToastContext from '../../../../../context/Toasts/ToastContext';
import ExpenseContext from '../../../../../context/Expense/expenseContext';

describe('AddExpense component', () => {
  test('Renders correctly', () => {
    const expensesList = new TestRenderer.create(
      (
        <ExpenseContext.Provider value={{ expenses: [] }}>
          <ToastContext.Provider value={{ messages: [] }}>
            <ExpensesList />
          </ToastContext.Provider>
        </ExpenseContext.Provider>
      )
    );
    expect(expensesList).toMatchSnapshot();
  });
});
