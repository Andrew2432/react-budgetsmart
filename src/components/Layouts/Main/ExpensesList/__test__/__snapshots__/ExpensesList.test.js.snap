// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpense component Renders correctly 1`] = `
<div
  className="card-panel indigo accent-2"
>
  <h4>
    Your Expenses
  </h4>
  <ul
    className="collection black-text"
  >
    <li
      className="collection-item"
    >
      No expenses added. Enter an expense
    </li>
  </ul>
</div>
`;
